app/routes.py

```diff
@@ -1,66 +1,49 @@
-# from pyramid.httpexceptions import HTTPBadRequest and other flask stuff
-# from pyramid.view import view_config
-# from pyramid.response import Response
-from flask import Response
-import six  # for compatibility Python 2 / 3
+from flask import Response, make_response, jsonify, abort
 from PIL import Image
 from app import app
-from app.helpers import check_color_channels
-from app.helpers import check_path
+from app.helpers.check_functions import check_color_channels
+from app.helpers.route import prefix_route
+from app.helpers import make_error_msg
 
-# Python2/3 compatibility
-if six.PY3:
-    from io import BytesIO
-else:
-    from StringIO import StringIO
+# add route prefix
+app.route = prefix_route(app.route, '/v4/color/')
 
 
 @app.route('/checker')
-@app.route('/checker/')
 def checker_page():
     """
     Just a route for the health check
     :return: OK with a 200 status code
     """
-    return "OK", 200
+    return make_response(jsonify({'success': True, 'message': 'OK'}))
 
 
-@app.route('/color/<r>,<g>,<b>/<filename>/', methods=['GET'])
-@app.route('/color/<r>,<g>,<b>/<filename>', methods=['GET'])
+@app.route('/<int:r>,<int:g>,<int:b>/<string:filename>', methods=['GET'])
 def color(r, g, b, filename):
     """
     This endpoint verifies the parameters of the request.
     BadRequests errors will be raised in case of not properly defined parameters.
-    If all parameters are well defined the colored image (symbol) is returned.
+    If all parameters are well defined, the colored image (symbol) is returned.
     :param r: integer value in the range between 0 and 255 for the red channel
     :param g: integer value in the range between 0 and 255 for the green channel
     :param b: integer value in the range between 0 and 255 for the blue channel
     :param filename: name of the file containing the image/symbol to be colored
     :return:
     """
-    check_color_channels(r, g, b)
-    r = int(r)
-    g = int(g)
-    b = int(b)
+    r, g, b = check_color_channels(r, g, b)
 
-    # the images are currently stored in a subfolder calles "maki".
-    # is it necessary to keep this subfolder, or is ../static/images/ ok?
     path = os.path.abspath(os.path.join(os.path.dirname(__file__), '../static/images/', filename))
+    if not os.path.isfile(path):
+        abort(make_error_msg(400, "The image to colorize doesn\'t exist"))
 
-    check_path(path)
-
-    mask = Image.open(path)
-
-    if mask.mode == 'P':
-        mask = mask.convert('RGBA')
-    img = Image.composite(Image.new("RGB", mask.size, (r, g, b)), mask, mask)
-    if six.PY3:
+    with Image.open(path) as mask:
+        if mask.mode == 'P':
+            mask = mask.convert('RGBA')
+        img = Image.composite(Image.new("RGB", mask.size, (r, g, b)), mask, mask)
         output = BytesIO()
-    else:
-        output = StringIO()
-    img.save(output, format='PNG')
+        img.save(output, format='PNG')
+        img.close()
```

Comment:
From my understanding of the pillow api docu the call of close() on an image created by new() or composite() is not necessary.

Reply:
probably right, but I am not 100% sure. And the close() won't hurt anybody and causes no errors. So I'd opt for keeping it at the Moment